### PR TITLE
Surgery Adjustments & Fixes

### PR DESCRIPTION
**Pull request by WhiteHusky**
**!!!** Might need some wiki updates.

I've sorted some stuff out with surgery.
Also,
```dm
find_working_surface_at_mob(mob/living/carbon/human/target, list/possible_surfaces_list)
```
Finds if the mob is on a structure, item, etc associated to values before returning the value for the matching item.
```dm
find_working_surface_at_mob_verbose(mob/living/carbon/human/target, list/possible_surfaces_list)
```
Similar to find_working_surface_at_mob() but returns a list containing both the value and the surface. `list("value" = value, "working_surface" = working_surface)`
```dm
target_to_mobpart(var/limb)
```
Converts a target like `LIMB_CHEST` to a mobpart like `UPPER_TORSO`
```dm
get_clothing_obstructing_target_from_user(mob/living/user, mob/living/carbon/human/target, list/armorlimits = null, list/exceptions = null)
```
Finds the piece of clothing obstructing the selected targeted limb from the user to the target else returns FALSE.
If armorlimits is provided, it checks if any of the armor stats in the clothing will exceed the armorlimits provided to return FALSE.
If exceptions are provided, it checks if the cover is in the list to return FALSE.
```dm
does_clothing_exceed_armor_limits(obj/item/clothing/cover, list/armorlimits)
```
Takes a cover and checks against the provided list of armor stats and returns TRUE if any of the armor stats in the cover exceeds the ones provided else FALSE
```dm
get_surface_medication_obstruction(mob/living/user, mob/living/carbon/human/target)
```
Returns the piece of cover that would obstruct medication from user to target. Otherwise FALSE.
```dm
check_if_ready_for_surgery(mob/living/target, mob/living/user, obj/item/tool)
```
Checks if the mob is ready for surgery and returns the surface stability or `FALSE`.
```dm
ready_for_surgery(mob/living/target, mob/living/user, obj/item/tool)
```
Checks if the mob is ready for surgery and if true, sets some preflight vars in the tool before returning `TRUE`, else returns `FALSE`
```dm
unready_surgery(obj/item/tool)
```
Abort the preflight and clear the vars for it.
```dm
do_surgery(mob/living/target = null, mob/living/user = null, obj/item/tool)
```
Attempt to do surgery and return `TRUE` if it was successfully started or at least failed due to something wrong with tool used while surgery can be currently done otherwise. Also can be ran with only a tool specified if the tool was preflighted with `ready_for_surgery()` (done to prevent finding a surface again), then it'll be cleared from preflight so it isn't necessary to run `unready_surgery()` afterwards.

So if some item wants to check if it needs to be involved in a surgery or not without causing a redundant check for surfaces to work on,
```dm
if(ready_for_surgery(my_target, my_user, src))
    do_surgery(tool = src)
else
    do_something_else()
```

# TODO
- [ ] Fix issue with cutting limbs failing silently.
- [ ] Update Wiki
:cl:
 * bugfix: Ghetto surgeries on unsuitable work surfaces no longer cause you to attack.
 * bugfix: Some ghetto surgery tools were fixed and work again.
 * bugfix: Ghetto face surgery with cable coils works if needed, otherwise starts teeth surgery. Expect to backtrack to fix the face if you screw up unless you have better tools however.
 * bugfix: You no longer stab people in the eyes if you're using ghetto surgery tools like forks or screwdrivers.
 * bugfix: Welding fuel is now used for fixing robo-limbs.
 * rscadd: You can now use painkillers to help perform surgeries on conscious patients.
 * rscadd: Surgery and Medication checks if the limb is obstructed by significant clothing.
 * rscadd: Moving or removing the structure to do surgery on will fail the surgery
 * rscadd: You are now punished if the skull or ribcage is not correctly mended back in place after cutting it.
 * rscdel: You can no longer do surgeries or apply meds on limbs through any significantly armored suit or helmet.
 * tweak: Surgeries will check if the patient is still asleep/on pain killers on every stage instead of doing it on the initial cut.
 * tweak: Reinforced tables gives you a better chance on ghetto surgeries.
 * tweak: Wooden tables are worse for doing surgery on
 * tweak: Autopsy is now a proper surgery procedure, but everything works the same. (Cut limb open and scan as always)
 * rscadd: Countless unknowns were killed in the making of this.

### CODE REVIEW

**Comment by DamianX** [comment]
What's so special about space suits, though? Would you do brain surgery on someone wearing a security helmet? Heart surgery through a bulletproof armor?

**Comment by WhiteHusky** [comment]
@DamianX Oh right. Yeah I'll add some logic for that.

**Comment by ShiftyRail** [comment]
>You can no longer do surgeries through space suits.

Nice, but please make an exception for plasmemes

**Comment by WhiteHusky** [comment]
@ShiftyRail ~~Good news, the new code I added will let that happened.
Bad news, it only works on plasmamen suits that are not armored. So uh, keep a spare suit for the plasmamen to change into first.~~
Nevermind, I got ahead of myself because I didn't notice the armor stats for the plasmaman suits on first glance.

**Comment by WhiteHusky** [comment]
I'm mixed if I really want to do make an exception for plasmamen, but I'll put it in.

**Comment by WhiteHusky** [comment]
I have put in the exception. Also adjusted the armor limit to 10 so you can medicate officers and anyone else who is moderately armored in their jumpsuit.

**Comment by ShiftyRail** [comment]
If you dislike the exception, make it a var at armour level and check for that.

File: code/game/objects/items/stacks/medical.dm [hunk 1]
```diff
@@ -25,14 +25,12 @@
...
-				to_chat(user, "<span class='warning'>You can't apply \the [src] through \the [H.head]!</span>")
-				return 1
-		else
-			if(H.wear_suit && istype(H.wear_suit,/obj/item/clothing/suit/space))
-				to_chat(user, "<span class='warning'>You can't apply \the [src] through \the [H.wear_suit]!</span>")
-				return 1
+		var/obj/item/clothing/cover = H.get_body_part_coverage(target_to_mobpart(user.zone_sel.selecting))
+		if(cover)
+			for (var/A in cover.armor)
+				if(cover.armor[A] > 5)
```
Comment (ShiftyRail) [inline code/game/objects/items/stacks/medical.dm]: You use a define instead of `5`

File: code/modules/surgery/surgery.dm [hunk 2]
```diff
@@ -26,14 +26,14 @@
...
 			return allowed_tools[T]
 	return 0
-
 /datum/surgery_step/proc/check_anesthesia(var/mob/living/carbon/human/target)
 	if(target.sleeping > 0 || target.stat)
 		return 1
 	if(!target.feels_pain())
 		return 1
-	if(prob(25)) // Pain is tolerable?  Pomf wanted this. - N3X
+	if(target.pain_numb || prob(target.pain_tolerance + 5)) // Pain is tolerable?  Pomf wanted this. - N3X | How about painkillers? - Carlen
```
Comment (ShiftyRail) [inline code/modules/surgery/surgery.dm]: Same thing, a define could be better

File: code/modules/surgery/surgery.dm [hunk 3]
```diff
@@ -97,10 +97,43 @@ proc/spread_germs_to_organ(datum/organ/external/E, mob/living/carbon/human/user)
...
 		return 0
+
+	// VOTE! Should surgery even proceed if there's a suit in the way?
+	var/mob/living/carbon/human/MH = M
+	var/obj/item/clothing/cover = MH.get_body_part_coverage(target_to_mobpart(user.zone_sel.selecting))
+	if(cover)
+		for (var/A in cover.armor)
+			if(cover.armor[A] > 5)
+				to_chat(user, "<span class='warning'>You can't use \the [tool] through \the [cover]!</span>")
+				return 1
```
Comment (ShiftyRail) [inline code/modules/surgery/surgery.dm]: `return TRUE` is the standard I think

File: code/modules/surgery/surgery.dm [hunk 4]
```diff
@@ -114,38 +147,38 @@ proc/do_surgery(mob/living/M, mob/living/user, obj/item/tool)
...
 					S.doing_surgery -= M
 				else
 					S.doing_surgery.Remove(null) //get rid of that now null reference
 				return	1	  												//don't want to do weapony things after surgery
 	if (user.a_intent == I_HELP)
-		to_chat(user, "<span class='warning'>You can't see any useful way to use [tool] on [M].</span>")
-		return 1
-	return 0
+		to_chat(user, "<span class='warning'>You want to help but you can't see any useful way to use [tool] on [M].</span>")
+	return 1
```
Comment (ShiftyRail) [inline code/modules/surgery/surgery.dm]: `return TRUE`

File: code/modules/surgery/tooth.dm [hunk 5]
```diff
@@ -26,8 +26,14 @@
...
-	return ..() && target.op_stage.face == 1
-
+	if(..() && target.op_stage.face == 1)
+		var/datum/organ/external/head/h = target.get_organ(target_zone)
+		if(h.disfigured) // Let's fix the face first
+			return 0
+		else
+			return 1
+	else
+		return 0
```
Comment (ShiftyRail) [inline code/modules/surgery/tooth.dm]: Replace the 0 and 1

File: code/modules/surgery/surgery.dm [hunk 6]
```diff
@@ -114,38 +147,38 @@ proc/do_surgery(mob/living/M, mob/living/user, obj/item/tool)
...
-							to_chat(user, "<span class='warning'>The patient is squirming around in pain!</span>")
-							M.emote("scream",,, 1)
 						M.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has had surgery [S.type] with \the [tool] failed by [user.name] ([user.ckey])</font>")
 						user.attack_log += text("\[[time_stamp()]\] <font color='red'>Failed surgery [S.type] with \the [tool] on [M.name] ([M.ckey])</font>")
 						log_attack("<font color='red'>[user.name] ([user.ckey]) used \the [tool] to fail the surgery type [S.type] on [M.name] ([M.ckey])</font>")
 						S.fail_step(user, M, user.zone_sel.selecting, tool)		//malpractice~
+						if(sleep_fail)
+							to_chat(user, "<span class='warning'>The patient is squirming around in pain!</span>")
+							M.emote("scream",,, 1)
+						else if (surface_stability != 100)
```
Comment (ShiftyRail) [inline code/modules/surgery/surgery.dm]: Could use a define

File: code/modules/surgery/surgery.dm [hunk 7]
```diff
@@ -97,10 +97,43 @@ proc/spread_germs_to_organ(datum/organ/external/E, mob/living/carbon/human/user)
...
+
+	// type path referencing surfaces that could be used
+	var/list/allowed_work_surfaces = list(
+		/obj/machinery/optable = 100,
+		/obj/structure/bed/roller/surgery = 100,
+		/obj/structure/bed/roller = 75,
+		/obj/structure/table/reinforced = 70,
+		/obj/structure/table = 66
+		)
+	var/surface_stability = 0
```
Comment (ShiftyRail) [inline code/modules/surgery/surgery.dm]: Could use a define

File: code/modules/mob/living/carbon/shock.dm [hunk 8]
```diff
@@ -35,6 +36,11 @@
 				if(organ.status & ORGAN_SPLINTED)
 					pain_level -= 25
 
+	pain_tolerance = pain_level * -1
+
+	if(pain_tolerance < 0)
```
Comment (ShiftyRail) [inline code/modules/mob/living/carbon/shock.dm]: Use `max(0, pain_level * -1)`

File: code/game/objects/items/stacks/medical.dm [hunk 9]
```diff
@@ -221,11 +210,7 @@
 				affecting.heal_damage(0, rand(heal_burn, heal_burn + 5))
 				use(1)
 		else
-			if(can_operate(H, user))        //Checks if mob is lying down on table for surgery
-				if(do_surgery(H,user,src))
-					return
-			else
-				to_chat(user, "<span class='notice'>[H]'s [affecting.display_name] is cut wide open, even a regenerative membrane won't do!</span>")
+			to_chat(user, "<span class='notice'>[H]'s [affecting.display_name] is cut wide open, even a regenerative membrane won't do!</span>")
```
Comment (ShiftyRail) [inline code/game/objects/items/stacks/medical.dm]: `\The [H]`

File: code/game/objects/items/stacks/medical.dm [hunk 10]
```diff
@@ -188,11 +181,7 @@
 				affecting.heal_damage(rand(heal_brute, heal_brute + 5), 0)
 				use(1)
 		else
-			if(can_operate(H, user))        //Checks if mob is lying down on table for surgery
-				if(do_surgery(H,user,src))
-					return
-			else
-				to_chat(user, "<span class='notice'>[H]'s [affecting.display_name] is cut wide open, even bioglue won't do!</span>")
+			to_chat(user, "<span class='notice'>[H]'s [affecting.display_name] is cut wide open, even bioglue won't do!</span>")
```
Comment (ShiftyRail) [inline code/game/objects/items/stacks/medical.dm]: `\The [H]`

File: code/game/objects/items/stacks/medical.dm [hunk 11]
```diff
@@ -128,11 +125,7 @@
 										"<span class='notice'>You salve the wounds on [M]'s [affecting.display_name].</span>" )
 				use(1)
 		else
-			if(can_operate(H, user))        //Checks if mob is lying down on table for surgery
-				if(do_surgery(H,user,src))
-					return
-			else
-				to_chat(user, "<span class='notice'>[H]'s [affecting.display_name] is cut wide open, you'll need more than some ointment!</span>")
+			to_chat(user, "<span class='notice'>[H]'s [affecting.display_name] is cut wide open, you'll need more than some ointment!</span>")
```
Comment (ShiftyRail) [inline code/game/objects/items/stacks/medical.dm]: `\The [H]`

File: code/game/objects/items/stacks/medical.dm [hunk 12]
```diff
@@ -97,11 +98,7 @@
 										"<span class='notice'>You place a bandaid over \the [W.desc] on [M]'s [affecting.display_name].</span>")
 				use(1)
 		else
-			if(can_operate(H, user))        //Checks if mob is lying down on table for surgery
-				if(do_surgery(H,user,src))
-					return
-			else
-				to_chat(user, "<span class='notice'>[H]'s [affecting.display_name] is cut wide open, you'll need more than a bandage!</span>")
+			to_chat(user, "<span class='notice'>[H]'s [affecting.display_name] is cut wide open, you'll need more than a bandage!</span>")
```
Comment (ShiftyRail) [inline code/game/objects/items/stacks/medical.dm]: `\The [H]`

File: code/modules/surgery/other.dm [hunk 13]
```diff
@@ -183,3 +183,39 @@
...
+	if(tool.target_name != target.name)
+		tool.target_name = target.name
+		tool.wdata = list()
+		tool.chemtraces = list()
+		tool.timeofdeath = null
+		to_chat(user, "<span class='warning'>A new patient has been registered. Purging data for previous patient.</span>")
+
+	tool.timeofdeath = target.timeofdeath
+	tool.add_data(affected)
+	user.visible_message("<span class='notice'>[user] scans the wounds on [target]'s [affected.display_name].</span>", "<span class='notice'>You scan the wounds on [target]'s [affected.display_name].</span>", "You hear a beep.")
```
Comment (ShiftyRail) [inline code/modules/surgery/other.dm]: `\The [user]`, `\the [target]`

File: code/__HELPERS/type2type.dm [hunk 14]
```diff
@@ -262,3 +262,19 @@ proc/tg_jointext(list/list, glue = ",")
 		finalNum = copytext(finalNum, 1, pos) + sep + copytext(finalNum, pos)
 
 	return finalNum
+
+/proc/target_to_mobpart(var/limb)
+	switch(limb)
+		if(LIMB_HEAD)		return HEAD
```
Comment (ShiftyRail) [inline code/__HELPERS/type2type.dm]: No one line ifs

**Comment by SonixApache** [comment]
remove exception for plasmemes

**Comment by ShiftyRail** [comment]
@SonixApache
>be plasmeme
>break a bone
>go to fix it
>die of immolation because doctors panic

Absolutely amazing

**Comment by SonixApache** [comment]
>be plasmeme
>break a bone
>go to disposals
>incinerator
>mass driver

problem solved
in all seriousness having to work your way around the suit adds some challenge to performing medical and adds a logical downside to playing as a fire hazard in a hazmat suit

**Comment by DamianX** [comment]
What is the purpose of preventing surgery through certain things? My immulsions? If that's the case, I wouldn't do surgery through ANYTHING, not just armor. I wouldn't keep my patient's fez on if I'm opening his skull. Is it a balance change? A counter-quality-of-life change? What is it?

**Comment by Pathid** [comment]
Jumpsuits being exempt makes sense as we can pretend they're being zipped open and rolled down. Anything else is iffy. Making the surgery room a pure N2 environment to operate on plasmamen doesn't sound that bad. Or more realistically give them leporazine and they're fine.

**Comment by ghost** [comment]
Just Like Take Some Dermaline Nigga

**Comment by WhiteHusky** [comment]
@DamianX I'm almost done with a round of tidying up, but my thought process was that if the piece of clothing is not well armored then it could be easily opened, but armor vests, space suits, etc might need to be properly removed to do surgery.

**Comment by WhiteHusky** [comment]
Plus having to remove everything is a bigger pain in the ass (dropped items everywhere) then just removing a space suit (at worst, the suit and mask)

Not to mention we're now checking if that part of the limb is obstructed so if someone got assblasted by a shotgun while wearing a vest, you're likely pulling fragments out of the arms which the vest does not protect and won't be in the way of the surgery.

File: __DEFINES/global.dm [hunk 15]
```diff
@@ -353,3 +353,13 @@ var/list/extraMiniMaps = list()
 var/list/holomap_markers = list()
 
 var/holomaps_initialized = 0
+
+//BRUISEPACKS AND SURGERY
+
+#define CAN_MEDICATE_THROUGH_ARMOR FALSE // Ditto
+#define MEDICAL_AID_ARMOR_LIMIT list(melee = 10, bullet = 10, laser = 10,energy = 10, bomb = 10, bio = 10, rad = 10) // Anything tougher than most or all specialized jumpsuits like virologists or security.
+#define PERCENT_SUITABLE_MEDICAL_WORKSPACE 100 // Anything less than a surgery operation table or similar is no good and warns the player
+#define CAN_MEDICATE_THROUGH_PLASMAMEN_SUITS TRUE // Magic I don't know.
```
Comment (ShiftyRail) [inline __DEFINES/global.dm]: I had the idea to make it a var at amour level, like `var/can_medicate_through` and set it to `true` for plasmamen, but do as you wish.

File: code/__HELPERS/unsorted.dm [hunk 16]
```diff
@@ -1630,3 +1630,34 @@ Game Mode config tags:
 	if(!istype(T2))
 		T2 = get_turf(B)
 	return sqrt(((T2.x - T1.x) ** 2) + ((T2.y - T1.y) ** 2))
+
+// For surgeries
+
+
+/proc/find_working_surface_at_mob(mob/living/carbon/human/target, list/possible_surfaces_list)
+	for (var/surface in possible_surfaces_list)
+		world << "Trying to find: [surface]"
```
Comment (ShiftyRail) [inline code/__HELPERS/unsorted.dm]: You'll want to remove this once you're finished

File: __DEFINES/global.dm (hunk 15 above)
Comment (WhiteHusky) [reply __DEFINES/global.dm]: That's a much more graceful way of doing that, yeah.
I'll put that in.

File: code/__HELPERS/unsorted.dm (hunk 16 above)
Comment (WhiteHusky) [reply code/__HELPERS/unsorted.dm]: Oops

File: code/WorkInProgress/autopsy.dm [hunk 17]
```diff
@@ -188,7 +188,7 @@
 	if(istype(usr,/mob/living/carbon))
 		// place the item in the usr's hand if possible
 		usr.put_in_hands(P)
-
+/* Moved to surgery/other.dm
```
Comment (DamianX) [inline code/WorkInProgress/autopsy.dm]: Let's get rid of this entirely if it's unused.

File: code/__HELPERS/unsorted.dm [hunk 18]
```diff
@@ -1193,8 +1193,8 @@ var/global/list/common_tools = list(
 	istype(W, /obj/item/weapon/bonesetter)
 	)
 
-//check if mob is lying down on something we can operate him on.
-/proc/can_operate(mob/living/carbon/M, mob/U)
+//check if mob is lying down on something we can operate him on. !!OBSOLETE!!
+/*/proc/can_operate(mob/living/carbon/M, mob/U)
```
Comment (DamianX) [inline code/__HELPERS/unsorted.dm]: Remove if unused

File: code/__HELPERS/unsorted.dm [hunk 19]
```diff
@@ -1630,3 +1630,39 @@ Game Mode config tags:
...
 	return sqrt(((T2.x - T1.x) ** 2) + ((T2.y - T1.y) ** 2))
+
+// For surgeries
+
+
+/proc/find_working_surface_at_mob(mob/living/carbon/human/target, list/possible_surfaces_list)
+	for (var/surface in possible_surfaces_list)
+		if (locate(surface, target.loc))
+			var/obj/structure/table/table = surface
+			if(istype(table, /obj/structure/table) && table.flipped)
```
Comment (DamianX) [inline code/__HELPERS/unsorted.dm]: You can omit the second argument to istype and it will be inferred

File: code/__HELPERS/unsorted.dm [hunk 20]
```diff
@@ -1630,3 +1630,39 @@ Game Mode config tags:
...
+				return 0
+			else
+				return possible_surfaces_list[surface]
+
+// Armor limits, if not null, will only find clothing with armor greater than the specified values via a list. Exceptions are clothing that should be ignored if found.
+/proc/get_clothing_obstructing_target_from_user(mob/living/user, mob/living/carbon/human/target, armorlimits = null, list/exceptions = null)
+	if(!istype(target, /mob/living/carbon/human))
```
Comment (DamianX) [inline code/__HELPERS/unsorted.dm]: You can omit the second argument to istype and it will be inferred

**Comment by DamianX** [comment]
>bugfix: Ghetto surgeries on unsuitable work surfaces no longer cause you to attack.

Was this really a bug?

**Comment by Exxion** [comment]
It wasn't, if memory serves

**Comment by Riekland** [comment]
did you know that there's a difference between cloth that you can cut through with a knife or a pair of scissors and a fucking pressure suit that's rated to both keep out the cold embrace of space and keep the warm embrace of boiling plasma in
nonhuman races should NOT have things easy

**Comment by WhiteHusky** [comment]
@DamianX @Exxion I was under the impression it was or at least wasn't the intent. Either way it was a bit annoying going between attempting to use a crowbar to pry open the skin or bashing their chest in.

Now if you use an unsuitable surface, the probability of you succeeding is changed instead of having a chance of violently killing them.

**Comment by WhiteHusky** [comment]
Do we still want to do surgery while on any intent except `HARM` or only on `HELP`? Because the surgery code was made with the former.

**Comment by gbasood** [comment]
every time someone mentions plasmamen having downsides and sawbones pops in for that thumbs down with zero discussion all I can think of is making plasmamen into literal reskins of humans

not to say he doesn't sometimes have valid points but geez

**Comment by gbasood** [comment]
also, on intents, it should probably continue to work only with HELP, because that's consistent with other behaviors already in game

**Comment by WhiteHusky** [comment]
Alright. Once I'm sorted with work stuffs I'll change that.

**Comment by WhiteHusky** [comment]
That should implement being no longer able to do surgery when not on the `HELP` intent but once I get back to my desktop to compile and test, I'll see how it fairs.

**Comment by WhiteHusky** [comment]
I think that's about it on what I want to tweak and adjust.
Anything else?

File: __DEFINES/global.dm [hunk 21]
```diff
@@ -353,3 +353,14 @@ var/list/extraMiniMaps = list()
 var/list/holomap_markers = list()
 
 var/holomaps_initialized = 0
+
+//BRUISEPACKS AND SURGERY
+
+#define CAN_MEDICATE_THROUGH_ARMOR FALSE // Ditto
+#define MEDICAL_AID_ARMOR_LIMIT list(melee = 10, bullet = 10, laser = 10,energy = 10, bomb = 10, bio = 10, rad = 10) // Anything tougher than most or all specialized jumpsuits like virologists or security.
```
Comment (Exxion) [inline __DEFINES/global.dm]: Make this a global var, not a define. Otherwise it's creating a new list every time this is used.

File: __DEFINES/global.dm [hunk 22]
```diff
@@ -353,3 +353,14 @@ var/list/extraMiniMaps = list()
...
+#define PERCENT_SUITABLE_MEDICAL_WORKSPACE 100 // Anything less than a surgery operation table or similar is no good and warns the player
+#define CAN_MEDICATE_THROUGH_PLASMAMEN_SUITS FALSE // Magic if you can work through them I don't know.
+#define CAN_DO_SURGERY_ON_DISARM_GRAB_INTENT FALSE // Can you do surgery while on DISARM or GRAB?
+#define UNMEDICATED_PAIN_TOLERANCE 5 // Without meds, the probability of doing surgery on a conscious patient
+// type path referencing surfaces that could be used for medical work
+#define ALLOWED_MEDICAL_WORK_SURFACES list(/obj/machinery/optable = 100,/obj/structure/bed/roller/surgery = 100,/obj/structure/bed/roller = 75,/obj/structure/table/reinforced = 70,/obj/structure/table = 66)
```
Comment (Exxion) [inline __DEFINES/global.dm]: This too

File: code/__HELPERS/unsorted.dm [hunk 23]
```diff
@@ -1630,3 +1616,49 @@ Game Mode config tags:
...
+				return possible_surfaces_list[surface]
+
+// Armor limits, if not null, will only find clothing with armor greater than the specified values via a list. Exceptions are clothing that should be ignored if found.
+/proc/get_clothing_obstructing_target_from_user(mob/living/user, mob/living/carbon/human/target, list/armorlimits = null, list/exceptions = null)
+	if(!istype(target))
+		return FALSE
+	var/obj/item/clothing/cover = target.get_body_part_coverage(target_to_mobpart(user.zone_sel.selecting))
+	if(cover)
+		if(exceptions)
+			for(var/exception in exceptions)
```
Comment (Exxion) [inline code/__HELPERS/unsorted.dm]: `is_type_in_list()`

**Review by Exxion — DISMISSED** [review]

File: code/__HELPERS/unsorted.dm (hunk 23 above)
Comment (WhiteHusky) [reply code/__HELPERS/unsorted.dm]: I knew there had to be a proc that did exactly that because while I was writing that I had the feeling it wasn't right.

**Comment by WhiteHusky** [comment]
Travis is having a bit of a fit.
https://travis-ci.org/vgstation-coders/vgstation13/builds/314147563#L642-L671

**Comment by WhiteHusky** [comment]
Added some code to fix the ribs back into place if you decide to go the bonegel-set-bonegel route which also fixes present fractures. So if you have a guy with broken ribs and internal organ damage,
`Cut > Clamp > Retract > Saw > Retract > Mend organs > Bone gel > Set > Bone gel > Close incision.`
But if the bones are not broken, you can simply retract and bone gel the ribs back into place.
_But if you don't properly close the ribcage, **you'll cause fractures**._

The reasoning was since you are fixing the bones, you'll probably be fixing the sawed off ribs in the process.
Plus being able to leave loose ribs in someone isn't kosher.

**Comment by DamianX** [comment]
tl;dr but if there are gameplay changes you might want to consider writing about them in http://ss13.moe/wiki/index.php/Surgery

**Comment by DamianX** [comment]
I don't think it's supposed to fail silently when a step doesn't succeed (due to pain or else), is it?

> You are beginning to cut off the unknown's head with the circular saw.
> 
> You are beginning to cut off the unknown's head with the circular saw.
> 
> You are beginning to cut off the unknown's head with the circular saw.
> 
> You are beginning to cut off the unknown's head with the circular saw.

**Comment by WhiteHusky** [comment]
@DamianX It shouldn't be. If it fails it should fail with a message (and anything else.)

I'll have to look into it later and poke around to see what's happening.

**Comment by SonixApache** [comment]
ded?

**Comment by DamianX** [comment]
Conflicts and more than a month with no updates. Ask to reopen if you decide to finish it, but I suggest you do it incrementally in atomic PRs, to be honest.
